src/Marten/DocumentStore.cs

```diff
@@ -92,6 +91,14 @@ public virtual void Dispose()
             bulkInsertion.BulkInsert(documents, mode, batchSize);
         }
 
+        public void BulkInsertEnlistTransaction<T>(IReadOnlyCollection<T> documents,
```

Comment:
Rather than adding new methods for enlisting transaction, I think it would be better to add an optional param to the existing methods to deal with enlisting. Thoughts?

Reply:
@jeremydmiller Suggested to add new method.
https://github.com/JasperFx/marten/discussions/2148#discussioncomment-2787127

Reply:
Alight, didn't notice the discussion thread. Thanks for the heads up!

Reply:
Any other issues? Is it ready to be merged?

Reply:
Ian good with it, will approve the same. After approval from another review either by @jeremydmiller or @oskardudycz, will merge this PR.

Reply:
Not gonna lie and say I exactly remember the conversation, but an optional parameter is a breaking change (requires a recompile even though the member usage is technically the same). That was probably why I asked for an overload instead. I'm catching up now, and probably pulling this in.